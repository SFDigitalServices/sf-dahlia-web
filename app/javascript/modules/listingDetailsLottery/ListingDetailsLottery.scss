.modal {
  @apply max-w-0;
}

.modal__inner {
  @apply p-0;
}

.modal__close {
  z-index: 100;
}
