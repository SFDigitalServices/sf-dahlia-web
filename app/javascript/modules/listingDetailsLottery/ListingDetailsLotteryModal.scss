.lottery-modal {
    .lottery-results-form .input {
        border-radius: initial
    }
}