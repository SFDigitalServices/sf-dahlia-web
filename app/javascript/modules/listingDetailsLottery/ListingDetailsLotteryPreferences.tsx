import React from "react"
import { t } from "@bloom-housing/ui-components"
import dayjs from "dayjs"
import { RailsLotteryBucketsDetails } from "../../api/types/rails/listings/RailsLotteryBucketsDetails"
import { RailsListing } from "../listings/SharedHelpers"

export interface ListingDetailsLotteryPreferencesProps {
  listing: RailsListing
  lotteryBucketsDetails: RailsLotteryBucketsDetails
}

export const ListingDetailsLotteryPreferences = ({
  listing,
  lotteryBucketsDetails,
}: ListingDetailsLotteryPreferencesProps) => {
  return (
    <div className="text-tiny">
      <p className="py-4 text-sm">{t("lottery.buckets_intro")}</p>
      <h2 className="text-caps-underline">{t("lottery.housing_preferences")}</h2>
      <p className="border-b border-gray-450 mb-4 pb-4 text-gray-700 text-sm">
        {t("lottery.ranking_order_note")}
      </p>
      {lotteryBucketsDetails.lotteryBuckets
        .filter((bucket) => bucket.preferenceName !== "generalLottery")
        .map((bucket, index, arr) => (
          <>
            <h3
              className="font-sans font-semibold text-tiny tracking-wide uppercase"
              key={bucket.preferenceOrder}
            >
              {bucket.preferenceName}
            </h3>
            <p className="text-tiny">
              {t("lottery.up_to_x_units_available", { units: bucket.unitsAvailable })}
            </p>
            <p className="text-gray-700 text-tiny">
              {t("lottery.number_applicants_qualified_for_preference", {
                number: bucket.totalSubmittedApps,
              })}
            </p>
            <hr className={arr.length - 1 === index ? "mt-4" : "my-4"} />
          </>
        ))}
      <div className="bg-gray-100 border-b mb-4 p-4">
        <hr className="border-b-4 border-primary" />
      </div>
      {lotteryBucketsDetails.lotteryBuckets
        .filter((bucket) => bucket.preferenceName === "generalLottery")
        .map((bucket) => (
          <div key={bucket.preferenceOrder}>
            <h3 className="font-sans font-semibold text-tiny tracking-wide uppercase">
              {t("lottery.general_pool")}
            </h3>
            <p className="text-tiny">{t("lottery.any_remaining_units")}</p>
            <p className=" border-b pb-2 text-gray-700 text-tiny">
              {t("lottery.number_applicants_qualified_for_general_pool", {
                number: bucket.totalSubmittedApps,
              })}
            </p>
          </div>
        ))}
      <div className="bg-gray-100 mb-2 py-4 text-center">
        <a href="http://sfmohcd.org/housing-preference-programs" target="_blank">
          {t("lottery.read_about_preferences")}
        </a>
      </div>
      <div className="text-center">
        <p className="py-2">
          {t("lottery.results_from", {
            date: dayjs(listing.Lottery_Results_Date).format("MMM D, YYYY"),
          })}
        </p>
        <p className="py-2">
          <a href={listing.LotteryResultsURL} target="_blank">
            {t("lottery.lottery_results_pdf_link")}
          </a>
        </p>
        <p className="text-gray-700 py-2">
          {t("lottery.unsorted_results_note", {
            date: dayjs(listing.Lottery_Date).format("MMM D, YYYY"),
          })}
        </p>
      </div>
    </div>
  )
}
