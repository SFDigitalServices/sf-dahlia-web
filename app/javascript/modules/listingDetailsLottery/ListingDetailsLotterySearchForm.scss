.lottery-search-form {
    #search-form input {
        border-radius: initial
    }

    #search-form input::placeholder {
        @apply text-gray-750;
    }
}