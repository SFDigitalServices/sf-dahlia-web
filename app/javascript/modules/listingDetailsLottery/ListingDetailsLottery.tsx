import React, { useEffect, useState } from "react"
import {
  AppearanceSizeType,
  AppearanceStyleType,
  Button,
  Heading,
  Modal,
  t,
} from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { isLotteryComplete } from "../../util/listingUtil"
import { getLotteryBucketDetails } from "../../api/listingApiService"
import { RailsLotteryResult } from "../../api/types/rails/listings/RailsLotteryResult"
import { ListingDetailsLotteryModal } from "./ListingDetailsLotteryModal"
import { localizedFormat, renderInlineWithInnerHTML } from "../../util/languageUtil"

export interface ListingDetailsLotteryProps {
  listing: RailsListing
}

export const ListingDetailsLottery = ({ listing }: ListingDetailsLotteryProps) => {
  const [lotteryBucketDetails, setLotteryBucketDetails] = useState<RailsLotteryResult>()
  const [isModalOpen, setIsModalOpen] = useState(false)

  useEffect(() => {
    void getLotteryBucketDetails(listing.Id).then((lotteryBucketDetails) => {
      setLotteryBucketDetails(lotteryBucketDetails)
    })
  }, [listing.Id])

  return (
    isLotteryComplete(listing) && (
      <div className="border-b pt-4 text-center">
        <Heading className="mb-4" priority={4}>
          {t("lottery.lotteryResults")}
        </Heading>
        <p className="mb-4 text-sm uppercase">
          {localizedFormat(listing.Lottery_Results_Date, "LL")}
        </p>
        <div className="bg-gray-100 py-4">
          {listing.Lottery_Summary && (
            <div className="mb-3 mx-2 text-gray-700 text-sm">
              {renderInlineWithInnerHTML(listing.Lottery_Summary)}
            </div>
          )}
          <Button
            size={AppearanceSizeType.small}
            styleType={AppearanceStyleType.primary}
            onClick={() => {
              setIsModalOpen(true)
            }}
          >
            {t("listings.lottery.viewLotteryResults")}
          </Button>
        </div>

        {lotteryBucketDetails && (
          <Modal onClose={() => setIsModalOpen(false)} open={isModalOpen} title="">
            <ListingDetailsLotteryModal
              listing={listing}
              lotteryBucketDetails={lotteryBucketDetails}
            />
          </Modal>
        )}
      </div>
    )
  )
}
