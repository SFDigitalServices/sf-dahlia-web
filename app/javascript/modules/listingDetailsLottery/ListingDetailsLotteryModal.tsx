import React, { useState } from "react"
import { Field, Heading, Icon, IconFillColors, Modal, t } from "@bloom-housing/ui-components"
import { useForm } from "react-hook-form"
import { RailsLotteryBucketsDetails } from "../../api/types/rails/listings/RailsLotteryBucketsDetails"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsLotteryPreferences } from "./ListingDetailsLotteryPreferences"
import "./ListingDetailsLotteryModal.scss"
import { getLotteryResults } from "../../api/listingApiService"
import { RailsLotteryRanking } from "../../api/types/rails/listings/RailsLotteryRanking"
import { ListingDetailsLotteryRanking } from "./ListingDetailsLotteryRanking"
import { ListingDetailsLotteryModalFooter } from "./ListingDetailsLotteryModalFooter"

export enum STATES {
  INITIAL_STATE,
  LOADING,
  API_ERROR,
  INVALID_LOTTERY_NUMBER,
  SEARCH_RESULT_FOUND,
}

interface ListingDetailsLotteryModalProps {
  isOpen: boolean
  listing: RailsListing
  lotteryBucketDetails: RailsLotteryBucketsDetails
  onClose: () => void
}

export const ListingDetailsLotteryModal = ({
  isOpen,
  listing,
  lotteryBucketDetails,
  onClose,
}: ListingDetailsLotteryModalProps) => {
  // eslint-disable-next-line @typescript-eslint/unbound-method
  const { errors, handleSubmit, register } = useForm({ reValidateMode: "onSubmit" })
  const lotteryNumberField = "lotterySearchNumber"
  const [lotterySearchResult, setLotterySearchResult] = useState<RailsLotteryRanking>()

  const [lotteryModalStatus, setLotteryModalStatus] = useState<STATES>(STATES.INITIAL_STATE)

  const onModalClose = () => {
    setLotteryModalStatus(STATES.INITIAL_STATE)
    onClose()
  }

  const onSubmit = (data: { lotterySearchNumber: string }) => {
    const { lotterySearchNumber } = data
    setLotteryModalStatus(STATES.LOADING)

    void getLotteryResults(listing.Id, lotterySearchNumber).then((lotterySearchResults) => {
      if (!lotterySearchResults) {
        // if we didn't get a result, general search api error
        setLotteryModalStatus(STATES.API_ERROR)
        return
      }

      const { lotteryBuckets } = lotterySearchResults
      if (
        // if we got a result but there are no buckets or none of the buckets have pref
        // results, then we don't have a valid lottery number
        !lotteryBuckets ||
        lotteryBuckets.filter((bucket) => bucket.preferenceResults.length > 0).length === 0
      ) {
        setLotteryModalStatus(STATES.INVALID_LOTTERY_NUMBER)
      } else {
        setLotteryModalStatus(STATES.SEARCH_RESULT_FOUND)
        setLotterySearchResult(lotterySearchResults)
      }
    })
  }

  let content: JSX.Element
  switch (lotteryModalStatus) {
    case STATES.LOADING: {
      content = (
        <div className="text-center">
          <Icon size="large" symbol="spinner" />
        </div>
      )
      break
    }
    case STATES.INVALID_LOTTERY_NUMBER: {
      content = (
        <div className="my-6 text-gray-700">
          <p className="mb-2">{t("lottery.lotteryNumberNotFoundP2")}</p>
          <p className="mb-2">{t("lottery.lotteryNumberNotFoundP3")}</p>
        </div>
      )
      break
    }
    case STATES.SEARCH_RESULT_FOUND: {
      content = lotterySearchResult && (
        <ListingDetailsLotteryRanking lotteryRanking={lotterySearchResult} />
      )
      break
    }
    case STATES.INITIAL_STATE: {
      content = <ListingDetailsLotteryPreferences lotteryBucketsDetails={lotteryBucketDetails} />
      break
    }
  }

  const errorMessage =
    (errors[lotteryNumberField]?.type === "required" && t("lottery.lotteryNumberNotValid")) ||
    (lotteryModalStatus === STATES.INVALID_LOTTERY_NUMBER &&
      t("lottery.lotteryNumberNotFoundP1")) ||
    (lotteryModalStatus === STATES.API_ERROR && t("error.lotteryRankingSearch"))

  return (
    <Modal onClose={onModalClose} open={isOpen} title="">
      <div className="lottery-modal w-96">
        <header className="pb-4 text-center">
          <Heading>{t("lottery.lotteryResults")}</Heading>
          <h2 className="font-sans font-semibold text-sm uppercase">{listing.Name}</h2>
        </header>
        <form
          className="bg-gray-100 flex lottery-results-form mb-4 px-6 py-4"
          onSubmit={handleSubmit(onSubmit)}
        >
          <Field
            error={
              errors[lotteryNumberField] ||
              lotteryModalStatus === STATES.API_ERROR ||
              lotteryModalStatus === STATES.INVALID_LOTTERY_NUMBER
            }
            className="w-full"
            errorMessage={
              errorMessage ||
              (errors[lotteryNumberField]?.type === "required" &&
                t("lottery.lotteryNumberNotValid"))
            }
            register={register}
            name={lotteryNumberField}
            placeholder={t("lottery.enterLotteryNumber")}
            type="text"
            validation={{ required: true }}
          />
          <button className="bg-blue-600 h-12 mt-2 p-3">
            <Icon fill={IconFillColors.white} size="medium" symbol="right" />
          </button>
        </form>
        {content}
        <ListingDetailsLotteryModalFooter
          lotteryModalStatus={lotteryModalStatus}
          listing={listing}
        />
      </div>
    </Modal>
  )
}
