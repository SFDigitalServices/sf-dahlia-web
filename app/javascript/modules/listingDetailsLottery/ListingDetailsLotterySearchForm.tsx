import React, { useState } from "react"
import { Field, Heading, Icon, IconFillColors, t } from "@bloom-housing/ui-components"
import { useForm } from "react-hook-form"
import type { RailsLotteryResult } from "../../api/types/rails/listings/RailsLotteryResult"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsLotteryPreferences } from "./ListingDetailsLotteryPreferences"
import { getLotteryResults } from "../../api/listingApiService"
import { ListingDetailsLotteryRanking } from "./ListingDetailsLotteryRanking"
import { ListingDetailsLotterySearchFooter } from "./ListingDetailsLotterySearchFooter"
import "./ListingDetailsLotterySearchForm.scss"
import ErrorBoundary, { BoundaryScope } from "../../components/ErrorBoundary"

export enum LOTTERY_SEARCH_FORM_STATUS {
  INITIAL_STATE,
  LOADING,
  API_ERROR,
  INVALID_LOTTERY_NUMBER,
  SEARCH_RESULT_FOUND,
}

interface ListingDetailsLotterySearchFormProps {
  listing: RailsListing
  lotteryBucketDetails: RailsLotteryResult
}

export const ListingDetailsLotterySearchForm = ({
  listing,
  lotteryBucketDetails,
}: ListingDetailsLotterySearchFormProps) => {
  // eslint-disable-next-line @typescript-eslint/unbound-method
  const { errors, handleSubmit, register } = useForm({ reValidateMode: "onSubmit" })
  const lotteryNumberField = "lotterySearchNumber"
  const [lotterySearchResult, setLotterySearchResult] = useState<RailsLotteryResult>()
  const [lotteryFormStatus, setLotteryFormStatus] = useState<LOTTERY_SEARCH_FORM_STATUS>(
    LOTTERY_SEARCH_FORM_STATUS.INITIAL_STATE
  )

  const onSubmit = (data: { lotterySearchNumber: string }) => {
    const lotterySearchNumber = data.lotterySearchNumber
    setLotteryFormStatus(LOTTERY_SEARCH_FORM_STATUS.LOADING)

    void getLotteryResults(listing.Id, lotterySearchNumber).then((lotterySearchResults) => {
      if (!lotterySearchResults) {
        // if we didn't get a result, general search api error
        setLotteryFormStatus(LOTTERY_SEARCH_FORM_STATUS.API_ERROR)
        return
      }

      const { lotteryBuckets } = lotterySearchResults
      if (
        // if we got a result but there are no buckets or none of the buckets have pref
        // results, then we don't have a valid lottery number
        !lotteryBuckets ||
        lotteryBuckets.filter((bucket) => bucket.preferenceResults.length > 0).length === 0
      ) {
        setLotteryFormStatus(LOTTERY_SEARCH_FORM_STATUS.INVALID_LOTTERY_NUMBER)
      } else {
        setLotteryFormStatus(LOTTERY_SEARCH_FORM_STATUS.SEARCH_RESULT_FOUND)
        setLotterySearchResult(lotterySearchResults)
      }
    })
  }

  let content: JSX.Element
  switch (lotteryFormStatus) {
    case LOTTERY_SEARCH_FORM_STATUS.INITIAL_STATE:
    case LOTTERY_SEARCH_FORM_STATUS.API_ERROR: {
      content = <ListingDetailsLotteryPreferences lotteryBucketsDetails={lotteryBucketDetails} />
      break
    }
    case LOTTERY_SEARCH_FORM_STATUS.LOADING: {
      content = (
        <div className="pb-4 text-center">
          <Icon size="large" symbol="spinner" />
        </div>
      )
      break
    }
    case LOTTERY_SEARCH_FORM_STATUS.INVALID_LOTTERY_NUMBER: {
      content = (
        <div className="my-6 px-8 text-gray-700">
          <p className="mb-2">{t("lottery.lotteryNumberNotFoundP2")}</p>
          <p className="mb-2">{t("lottery.lotteryNumberNotFoundP3")}</p>
        </div>
      )
      break
    }
    case LOTTERY_SEARCH_FORM_STATUS.SEARCH_RESULT_FOUND: {
      content = lotterySearchResult && (
        <ListingDetailsLotteryRanking lotteryResult={lotterySearchResult} />
      )
      break
    }
  }

  const errorMessage =
    (["required", "pattern"].includes(String(errors[lotteryNumberField]?.type)) &&
      t("lottery.lotteryNumberNotValid")) ||
    (lotteryFormStatus === LOTTERY_SEARCH_FORM_STATUS.INVALID_LOTTERY_NUMBER &&
      t("lottery.lotteryNumberNotFoundP1")) ||
    (lotteryFormStatus === LOTTERY_SEARCH_FORM_STATUS.API_ERROR && t("error.lotteryRankingSearch"))

  return (
    <div className="lottery-search-form">
      <header className="pt-8 pb-4 text-center bg-white sticky top-0 z-50">
        <Heading>{t("lottery.lotteryResults")}</Heading>
        <h2 className="font-sans font-semibold text-xs uppercase">{listing.Name}</h2>
      </header>
      <p className="px-8 pb-4 text-xs">{t("lottery.bucketsIntro")}</p>
      <form
        id="search-form"
        className="bg-gray-100 flex mb-4 px-6 py-4"
        // eslint-disable-next-line @typescript-eslint/no-misused-promises
        onSubmit={handleSubmit(onSubmit)}
      >
        <Field
          error={!!errorMessage}
          className="w-full"
          errorMessage={errorMessage}
          register={register}
          name={lotteryNumberField}
          placeholder={t("lottery.enterLotteryNumber")}
          type="text"
          validation={{ pattern: /^\d+$/, required: true }}
          label={t("lottery.enterLotteryNumber")}
          labelClassName="sr-only"
        />
        <button
          className="bg-blue-500 h-12 p-3 translate"
          type="submit"
          aria-label={t("lottery.submitNumber")}
        >
          <Icon fill={IconFillColors.white} size="medium" symbol="right" />
        </button>
      </form>
      <ErrorBoundary boundaryScope={BoundaryScope.component}>
        <div
          aria-live="polite"
          aria-busy={lotteryFormStatus === LOTTERY_SEARCH_FORM_STATUS.LOADING}
        >
          {content}
        </div>
      </ErrorBoundary>
      <ListingDetailsLotterySearchFooter
        lotterySearchFormStatus={lotteryFormStatus}
        listing={listing}
      />
    </div>
  )
}
