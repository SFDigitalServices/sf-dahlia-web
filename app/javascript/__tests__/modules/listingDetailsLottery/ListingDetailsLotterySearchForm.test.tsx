import React from "react"
import { ListingDetailsLotterySearchForm } from "../../../modules/listingDetailsLottery/ListingDetailsLotterySearchForm"
import { lotteryCompleteRentalListing } from "../../data/RailsRentalListing/listing-rental-lottery-complete"
import { lotteryResultRentalThree } from "../../data/RailsLotteryResult/lottery-result-rental-three"
import userEvent from "@testing-library/user-event"
import { render, cleanup } from "@testing-library/react"
import { lotteryResultRentalInvalidLotteryNumber } from "../../data/RailsLotteryResult/lottery-result-rental-invalid-lottery-number"
import { renderAndLoadAsync } from "../../__util__/renderUtils"

const axios = require("axios")

jest.mock("axios")

describe("ListingDetailsLotteryModal", () => {
  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
    jest.resetAllMocks()
  })

  it("displays initial view with form and listing preferences", async () => {
    axios.get.mockResolvedValue({ data: lotteryResultRentalThree })

    const { findByText, asFragment } = render(
      <ListingDetailsLotterySearchForm
        listing={lotteryCompleteRentalListing}
        lotteryBucketDetails={lotteryResultRentalThree}
      />
    )

    expect(await findByText("Lottery Results")).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })

  it("displays error when user submits form with empty lottery number", async () => {
    const user = userEvent.setup()
    const { findByText, getByRole } = render(
      <ListingDetailsLotterySearchForm
        listing={lotteryCompleteRentalListing}
        lotteryBucketDetails={lotteryResultRentalThree}
      />
    )

    await user.click(getByRole("button"))

    expect(await findByText("Please enter a valid lottery number.")).toBeDefined()
  })

  it("displays error when user submits form with non numeric lottery number", async () => {
    const user = userEvent.setup()
    const { findByText, getByRole } = render(
      <ListingDetailsLotterySearchForm
        listing={lotteryCompleteRentalListing}
        lotteryBucketDetails={lotteryResultRentalThree}
      />
    )

    const input = getByRole("textbox")
    await userEvent.type(input, "123abc")
    await user.click(getByRole("button"))

    expect(await findByText("Please enter a valid lottery number.")).toBeDefined()
  })

  it("displays three results when lottery number found", async () => {
    axios.get.mockResolvedValue({ data: lotteryResultRentalThree })

    const user = userEvent.setup()
    const { findByText, getByRole } = await renderAndLoadAsync(
      <ListingDetailsLotterySearchForm
        listing={lotteryCompleteRentalListing}
        lotteryBucketDetails={lotteryResultRentalThree}
      />
    )

    const input = getByRole("textbox")
    await userEvent.type(input, "123")
    await user.click(getByRole("button"))

    expect(await findByText("Your preference ranking")).toBeDefined()
    expect(await findByText("Displaced Tenant Housing Preference (DTHP)")).toBeDefined()
    expect(await findByText("Certificate of Preference (COP)")).toBeDefined()
    expect(await findByText("Live or Work in San Francisco Preference")).toBeDefined()
  })

  it("displays results when a lotteryNumber is given initially", async () => {
    axios.get.mockResolvedValue({ data: lotteryResultRentalThree })

    const { findByText } = await renderAndLoadAsync(
      <ListingDetailsLotterySearchForm
        listing={lotteryCompleteRentalListing}
        lotteryBucketDetails={lotteryResultRentalThree}
        lotteryNumber="123"
      />
    )

    expect(await findByText("Your preference ranking")).toBeDefined()
    expect(await findByText("Displaced Tenant Housing Preference (DTHP)")).toBeDefined()
    expect(await findByText("Certificate of Preference (COP)")).toBeDefined()
    expect(await findByText("Live or Work in San Francisco Preference")).toBeDefined()
  })

  it("displays error when invalid lottery number", async () => {
    axios.get.mockResolvedValue({ data: lotteryResultRentalInvalidLotteryNumber })

    const user = userEvent.setup()
    const { findByText, getByRole } = await renderAndLoadAsync(
      <ListingDetailsLotterySearchForm
        listing={lotteryCompleteRentalListing}
        lotteryBucketDetails={lotteryResultRentalThree}
      />
    )

    const input = getByRole("textbox")
    await userEvent.type(input, "123")
    await user.click(getByRole("button"))

    expect(await findByText("The number you entered was not found.")).toBeDefined()
  })

  it("displays error when api error", async () => {
    axios.get.mockResolvedValue({ data: null })

    const user = userEvent.setup()
    const { findByText, getByRole } = await renderAndLoadAsync(
      <ListingDetailsLotterySearchForm
        listing={lotteryCompleteRentalListing}
        lotteryBucketDetails={lotteryResultRentalThree}
      />
    )

    const input = getByRole("textbox")
    await userEvent.type(input, "123")
    await user.click(getByRole("button"))

    expect(
      await findByText("We seem to be having a connection issue. Please try your search again.")
    ).toBeDefined()
  })
})
