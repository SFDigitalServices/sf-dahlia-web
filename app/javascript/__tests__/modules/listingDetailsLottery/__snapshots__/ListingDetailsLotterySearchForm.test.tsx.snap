// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsLotteryModal displays initial view with form and listing preferences 1`] = `
<div>
  <header
    className="pt-8 pb-4 text-center bg-white sticky top-0 z-50"
  >
    <h1
      className="undefined undefined"
    >
      Lottery Results
    </h1>
    <h2
      className="font-sans font-semibold text-sm uppercase"
    >
      855 Brannan Units 318
    </h2>
  </header>
  <form
    className="bg-gray-100 flex mb-4 px-6 py-4"
    id="search-form"
    onSubmit={[Function]}
  >
    <div
      className="field w-full"
    >
      <label
        className="label"
        htmlFor="lotterySearchNumber"
      />
      <div
        className="control"
      >
        <input
          aria-describedby="lotterySearchNumber"
          aria-invalid={false}
          className="input"
          id="lotterySearchNumber"
          name="lotterySearchNumber"
          placeholder="Enter Your Lottery Number"
          type="text"
        />
      </div>
    </div>
    <button
      className="bg-blue-600 h-12 mt-2 p-3"
    >
      <span
        className="ui-icon ui-medium"
        data-test-id={null}
      >
        <svg
          fill="#ffffff"
          viewBox="0 0 32 32"
        >
          <path
            d="M7.467 30.133c-0.4 0.4-0.4 1.2 0 1.6 0 0 0 0 0 0v0c0.4 0.4 1.2 0.4 1.6 0 0 0 0 0 0 0l14-14-1.467-1.733-14.133 14.133z"
          />
          <path
            d="M7.467 1.867l15.733 15.6 1.6-1.6-1.6-1.6-14.133-14c-0.4-0.4-1.067-0.4-1.6 0-0.4 0.533-0.4 1.2 0 1.6z"
          />
          <path
            d="M24.8 16v0 0z"
          />
        </svg>
      </span>
    </button>
  </form>
  <div
    aria-busy={false}
    aria-live="polite"
  >
    <div
      className="text-tiny"
    >
      <p
        className="px-8 pb-4 text-sm"
      >
        Lottery results are divided into multiple lists. Please see below for general information about each preference. For your specific results, please enter your lottery number.
      </p>
      <header
        className="px-8"
      >
        <h2
          className="undefined text-caps-underline"
        >
          Housing Preferences
        </h2>
        <p
          className="border-b border-gray-450 mb-4 pb-4 text-gray-700 text-sm"
        >
          Ranking in these lists is considered in the order shown here.
        </p>
      </header>
      <div
        className="px-8 "
      >
        <h3
          className="undefined font-sans font-semibold text-tiny tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        <p
          className="text-tiny"
        >
          Up to 1 units
        </p>
        <p
          className="text-gray-700 text-tiny"
        >
          1 applicants are on this list.
        </p>
      </div>
      <hr
        className="my-4 mx-8"
      />
      <div
        className="px-8 "
      >
        <h3
          className="undefined font-sans font-semibold text-tiny tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          className="text-tiny"
        >
          Up to 1 units
        </p>
        <p
          className="text-gray-700 text-tiny"
        >
          4 applicants are on this list.
        </p>
      </div>
      <hr
        className="my-4 mx-8"
      />
      <div
        className="px-8 "
      >
        <h3
          className="undefined font-sans font-semibold text-tiny tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          className="text-tiny"
        >
          Up to 1 units
        </p>
        <p
          className="text-gray-700 text-tiny"
        >
          905 applicants are on this list.
        </p>
      </div>
      <hr
        className="mt-4"
      />
      <div
        className="bg-gray-100 border-b mb-4 p-4"
      >
        <hr
          className="border-b-4 border-primary"
        />
      </div>
      <div
        className="px-8"
      >
        <h3
          className="undefined font-sans font-semibold text-tiny tracking-wide uppercase"
        >
          General Pool
        </h3>
        <p
          className="mb-1 text-tiny"
        >
          Any remaining units
        </p>
        <p
          className="text-gray-700 text-tiny"
        >
          193 applicants are on this list. Please note, preference holders will be considered for all units first.
        </p>
      </div>
      <hr
        className="mt-4"
      />
    </div>
  </div>
  <div
    className="bg-gray-100 py-4 text-center"
  >
    <a
      href="http://sfmohcd.org/housing-preference-programs"
      target="_blank"
    >
      Read About Housing Preferences
    </a>
  </div>
  <div
    className="p-4 text-center text-tiny"
  >
    <p
      className="py-2"
    >
      Results from January 26, 2022
    </p>
    <p
      className="py-2"
    >
      Complete lottery results will be posted in pdf format on January 26, 2022
    </p>
    <p
      className="text-gray-700 py-2"
    >
      This PDF contains the unsorted lottery results from the housing lottery held on January 25, 2022.
    </p>
  </div>
</div>
`;
